stl/inc/algorithm

```diff
@@ -5768,6 +5768,19 @@ namespace ranges {
         }
 
         if constexpr (bidirectional_iterator<_It>) {
+#if _USE_STD_VECTOR_ALGORITHMS
+            using _Elem = iter_value_t<_It>;
+
+            if constexpr (_Iterator_is_contiguous<_It> && sized_sentinel_for<_Se, _It>
+                          && conjunction_v<_Is_trivially_swappable<_Elem>, negation<is_volatile<_Elem>>>) {

```

Comment:
https://github.com/microsoft/STL/blob/da7307d68dbf39a805372e91a740393cbf06595c/stl/inc/algorithm#L5572

Also, `ranges::reverse` is using `_Is_trivially_ranges_swappable`. Because `_Rotate_unchecked()`'s normal codepath calls `_Reverse_common()`, I think we should consistently use `_Is_trivially_ranges_swappable` here.

I observe that using `_Is_trivially_ranges_swappable` cures the ADL tests - I believe because it short-circuits for scalar non-enums, and `holder<incomplete>*` is such a scalar.